futures-test/src/task/executor/noop.rs

```diff
@@ -0,0 +1,56 @@
+use futures_core::future::FutureObj;
+use futures_core::task::{Executor, SpawnObjError};
+use std::cell::UnsafeCell;
+
+/// An implementation of [`Executor`][futures_core::task::Executor] that
+/// discards spawned futures when used.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(async_await, futures_api)]
+///
+/// use futures::task::ExecutorExt;
+/// use futures_test::task::{panic_context, executor};
+///
+/// let mut cx = panic_context();
+/// let mut executor = executor::Noop::new();
+/// let cx = &mut cx.with_executor(&mut executor);
```

Comment:
Two issues with `Context`'s current APIs here:

The first (that's repeated in most of the examples) is that `with_executor` and `with_waker` take `&mut self`, that's useful for being able to create a sub-context with a different executor inside a `poll` function, but it is what makes the first of these three lines necessary. If `with_executor` instead took `self` by value we should be able to merge the first and third lines into

```rust
let cx = &mut panic_context().with_executor(&mut executor);
```

Second, `with_executor` requires `E: Sized` so we can't use the unsized `Noop::executor_mut()`, if we removed this bound then we can merge the second and third lines into

```rust
let cx = &mut cx.with_executor(executor::Noop::executor_mut());
```